gramps/gen/utils/alive.py

```diff
@@ -89,234 +96,420 @@ def __init__(
+                    death_found,
+                    explain_birth,
+                    explain_death,
+                )
+
+            if isinstance(class_or_handle, Person):
+                thisperson = class_or_handle
+            elif isinstance(class_or_handle, str):
+                thisperson = self.db.get_person_from_handle(class_or_handle)
             else:
-                birth_date = death_date
-            explain = _("death date")
-
-        if not death_date and birth_date:
-            # person died more than MAX after current year
-            death_date = birth_date.copy_offset_ymd(year=self.MAX_AGE_PROB_ALIVE)
-            explain = _("birth date")
-
-        if death_date and birth_date:
-            return (birth_date, death_date, explain, person)  # direct self evidence
-
-        # Neither birth nor death events are available. Try looking
-        # at siblings. If a sibling was born more than X years past,
-        # or more than Z future, then probably this person is
-        # not alive. If the sibling died more than X years
-        # past, or more than X years future, then probably not alive.
-
-        family_list = person.get_parent_family_handle_list()
-        for family_handle in family_list:
-            family = self.db.get_family_from_handle(family_handle)
-            if family is None:
-                continue
-            for child_ref in family.get_child_ref_list():
-                child_handle = child_ref.ref
-                child = self.db.get_person_from_handle(child_handle)
-                if child is None:
+                thisperson = None
+
+            if not thisperson:
+                LOG.debug("    get_person_bd(): null person called")
+                return (
+                    birth_date,
+                    death_date,
+                    death_found,
+                    explain_birth,
+                    explain_death,
+                )
+            # is there an actual death record?  Even if yes, there may be no date,
+            # in which case the EMPTY date is reported for the event.
+            death_ref = thisperson.get_death_ref()
+            if death_ref and death_ref.get_role().is_primary():
+                evnt = self.db.get_event_from_handle(death_ref.ref)
+                if evnt:
+                    death_found = True
+                    dateobj = evnt.get_date_object()
+                    if dateobj and dateobj.is_valid():
+                        death_date = dateobj
+                        explain_death = _("date")
+
+            # at this stage death_date is None or a valid date.
+            # death_found is true if thisperson is known to be dead,
+            #        whether or not a date was found.
+            # If we have no death_date then look for fallback even such as Burial.
+            # These fallbacks are fairly good indications that someone's not alive.
+            # If that date itself is not valid, it means we know they are dead
```

Comment:
```suggestion
            # If death_date itself is not valid, it means we know they are dead
```
I think this is the date you are talking about.

Reply:
At that stage I am referring to the fallback date being invalid. What I meant to says was...
"If the fallback death event does not have a valid date then..."
The comment seems rather self-evident now, but I initially wrote it in the context that the previous code would scan all fallback events and the last event would apply - potentially overwriting a good date with Today().

Reply:
That comment is clearer and makes sense given the proceeding sentence.

Reply:
Sorry to be thick, but am I able to change my comment in the code? it is not obvious from this UI view how I would do it.

Reply:
I'm not a github expert, but I think you need to edit the file in your branch to make the change.
That's essentially what happened in the background when you accepted my earlier suggestion:
![image](https://github.com/user-attachments/assets/b8104e04-96ce-4150-92b9-887a0633b7fc)
https://github.com/CameronD73/grampsf/commits/probablyAlive-0013443/

Reply:
OK done that.